account_invoice_alternate_payer/models/account_move.py

```diff
@@ -0,0 +1,161 @@
+# Copyright 2018 Eficent Business and IT Consulting Services, S.L.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from contextlib import contextmanager
+
+from odoo import _, api, fields, models
+from odoo.tools import float_is_zero
+
+
+class AccountMove(models.Model):
+
+    _inherit = "account.move"
+
+    alternate_payer_id = fields.Many2one(
+        "res.partner",
+        string="Alternate Payer",
+        readonly=True,
+        states={"draft": [("readonly", False)]},
+        help="If set, this will be the partner that we expect to pay or to "
+        "be paid by. If not set, the payor is by default the "
+        "commercial",
+    )
+
+    @api.depends("commercial_partner_id", "alternate_payer_id")
+    def _compute_bank_partner_id(self):
+        super_moves = self.filtered(
+            lambda r: not r.alternate_payer_id or not r.is_outbound()
+        )
+        for move in self - super_moves:
+            if move.is_outbound() and move.alternate_payer_id:
+                move.bank_partner_id = move.alternate_payer_id
+        return super(
+            AccountMove,
+            super_moves,
+        )._compute_bank_partner_id()
+
+    @api.onchange("alternate_payer_id")
+    def _onchange_alternate_payer_id(self):
+        return self._onchange_partner_id()
+
+    @contextmanager
+    def _sync_dynamic_line(
+        self,
+        existing_key_fname,
+        needed_vals_fname,
+        needed_dirty_fname,
+        line_type,
+        container,
+    ):
+        with super()._sync_dynamic_line(
+            existing_key_fname,
+            needed_vals_fname,
+            needed_dirty_fname,
+            line_type,
+            container,
+        ):
+            if line_type == "payment_term" and container.get("records").filtered(
+                lambda x: x.alternate_payer_id
+            ):
+                for invoice in container.get("records").filtered(
```

Comment:
Could you store `container.get("records").filtered(lambda x: x.alternate_payer_id)` in a variable so that you don't have to recalculate it on the `for` loop?

Reply:
Done!